### PR TITLE
[CF-250] Delete failed instances on dst

### PR DESCRIPTION
**Pull request by SVilgelm**
Delete failed instances on dst and restore state of vms on src

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:811: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:811: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:811: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:811: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:811: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:811: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:811: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:812: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:812: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:811: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:808: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:808: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferry/cfglib.py [hunk 1]
```diff
@@ -744,6 +744,15 @@
                     "the ports must be different."),
 ]
 
+rollback_group = cfg.OptGroup(name='rollback', title='Rollback settings')
+
+rollback_opts = [
+    cfg.BoolOpt('keep_migrated_vms', default=True,
+                help="Don not remove successfully migrated vms from dst and "
```
Comment (roman-verchikov) [inline cloudferry/cfglib.py]: Typo in "Don not": should be "Do not"

File: cloudferry/cfglib.py [hunk 2]
```diff
@@ -744,6 +744,15 @@
...
+                     "don not restore the  statuses of those vms on src "
```
Comment (roman-verchikov) [inline cloudferry/cfglib.py]: Same here

**Comment by roman-verchikov** [comment]
please update according to comment

File: cloudferry/cfglib.py (hunk 1 above)
Comment (SVilgelm) [inline cloudferry/cfglib.py]: Done

File: cloudferry/cfglib.py (hunk 2 above)
Comment (SVilgelm) [inline cloudferry/cfglib.py]: Done

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:808: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:808: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:808: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:808: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3474/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:807: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:807: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
Merge check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3481/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry_devlab.tests.rollback_scenario_generation
cloudferry_devlab/cloudferry_devlab/tests/rollback_scenario_generation.py:102: 
[E1101(no-member), RollbackScenarioGeneration.generate_exception_task_in_random_point] 
Instance of 'list' has no 'iteritems' member
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:807: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry_devlab.tests.rollback_scenario_generation
cloudferry_devlab/cloudferry_devlab/tests/rollback_scenario_generation.py:102: 
[E1101(no-member), RollbackScenarioGeneration.generate_exception_task_in_random_point] 
Instance of 'list' has no 'iteritems' member
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:807: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:807: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:807: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1

**Pull request merged**